<div class="alert alert-block alert-info">IAB303 - Business Intelligence - <a href="0%20-%20IAB303%20Overview.ipynb">overview</a></div>

## LECTURE :: Augmenting human intelligence in business

Imagine:
* You've just started a new job
* Marketing manager drops into your office with a problem: Our competition is overtaking us. Our customers seem to be turning away from our product. We don't know why. Can you help?

Consider:
* How significant is this problem for the business?
* What information do you need to address this concern?
* How do you go about obtaining this information?

### Human Intelligence as a starting point

* How does the Marketing manager describe the problem?
* What does their gut feeling, their intuition tell them?
* At a guess, why do they think this happening?

* The importance of intuition
* The risks of intuition

#### What value is Data Analytics for this scenario?

## Example

Say we want to find out what customers have been saying about our product that we are selling on Amazon...

[Data from Kaggle](https://www.kaggle.com/bittlingmayer/amazonreviews)

In [52]:
# Load 400,000 Amazon reviews (out of total dataset of 4 million)
pathToData = "../../data/kaggle-amazonreviews-test.txt"

# How big is this file?
import os
fileSize = os.stat(pathToData).st_size / (1024*1024)
print("File size: ",fileSize, "MB")

File size:  169.15911960601807 MB


In [53]:
# Now lets define a function to load the reviews
def readFileToList(fileName):
    file = open(fileName)
    rawtext = file.read()
    file.close()
    return rawtext.split("\n")

# Read the data into a List and assign to variable
reviewData = readFileToList(pathToData)

# Now lets check how many items in the list
numReviews = len(reviewData)

print("Number of reviews: ",numReviews)
# Why is this not 400k?

Number of reviews:  400001


In [54]:
# Let's look at the first and last reviews
print("First: ",reviewData[0]) # Why 0?
print("Last:",reviewData[-1])

First:  __label__2 Great CD: My lovely Pat has one of the GREAT voices of her generation. I have listened to this CD for YEARS and I still LOVE IT. When I'm in a good mood it makes me feel better. A bad mood just evaporates like sugar in the rain. This CD just oozes LIFE. Vocals are jusat STUUNNING and lyrics just kill. One of life's hidden gems. This is a desert isle CD in my book. Why she never made it big is just beyond me. Everytime I play this, no matter black, white, young, old, male, female EVERYBODY says one thing "Who was that singing ?"
Last: 


In [55]:
# last 2 items
reviewData[-2]

"__label__1 Comedy Scene, and Not Heard: This DVD will be a disappointment if you get it hoping to see some substantial portion of the acts of the various comics listed on the cover. All you get here are snippets of performance, at best. The rest is just loose-leaf reminiscence about the good old days in Boston, in the early 80's, when a lot of comics were hanging out together and getting their start.It's like a frat house reunion. There's a lot of lame nostalgia. There are quite a few guffaws recalling jokes (practical and otherwise)perpetrated - back then. But you had to have been there to appreciate all the basically good ol' boy camaraderie. If you weren't actually a part of that scene, all this joshing and jostling will fall flat.If you want to actually hear some of these comics' routines - you will have to look elsewhere."

In [56]:
# Sample 8 reviews
reviewData[0:-1:50000]

['__label__2 Great CD: My lovely Pat has one of the GREAT voices of her generation. I have listened to this CD for YEARS and I still LOVE IT. When I\'m in a good mood it makes me feel better. A bad mood just evaporates like sugar in the rain. This CD just oozes LIFE. Vocals are jusat STUUNNING and lyrics just kill. One of life\'s hidden gems. This is a desert isle CD in my book. Why she never made it big is just beyond me. Everytime I play this, no matter black, white, young, old, male, female EVERYBODY says one thing "Who was that singing ?"',
 "__label__2 remakes rarely stink in my eyes, this is no exception: John Waters, I admire your work, but this film almost surpasses your film. John Travolta in drag? Who would've thought! The whole cast shines in this film-now I want to see the original again and the play. I wouldn't expect so many people and critics to enjoy this but they did! The music numbers esp. are really good, some actors I never imagined could sing, surprised me, because

### Making sense

* How do we make meaning of this data?
* What intuitions might we have on the data?
* How do we test these intuitions?
* What do we need to do to make this raw data useful?

In [57]:
# What if we just want the CD reviews?
cdReviews = list(filter(lambda text: "CD" in text,reviewData))

# How many now?
print("CD reviews: ",len(cdReviews))

# Let's check the first few
cdReviews[0:3]

CD reviews:  21178


['__label__2 Great CD: My lovely Pat has one of the GREAT voices of her generation. I have listened to this CD for YEARS and I still LOVE IT. When I\'m in a good mood it makes me feel better. A bad mood just evaporates like sugar in the rain. This CD just oozes LIFE. Vocals are jusat STUUNNING and lyrics just kill. One of life\'s hidden gems. This is a desert isle CD in my book. Why she never made it big is just beyond me. Everytime I play this, no matter black, white, young, old, male, female EVERYBODY says one thing "Who was that singing ?"',
 '__label__1 A complete Bust: This game requires quicktime 5.0 to work...if you have a better version of quicktime (I have 7.5), it will ask you to install the quicktime available on the CD...if you click no, it will not let you play. So, I begrudgingly clicked yes on the third try, and it installed quicktime 5, THEN it tells me to please install the quicktime available on the disc. It KEPT telling me that, even after I uninstalled my version of

### Not so easy :(

* How long did take you to spot the problem in the first 3 reviews?
* How easy would it be to do with 21,178 reviews?
* What about millions?
* What is the computer good at?
* What is the computer bad at?
* What are humans good at?
* What are humans bad at?

## more about...

# Augmenting Human Intelligence

## than about...

# Business Intelligence 


<img src="ConceptualMap-Traditional.png"></img>

## Common (traditional) approaches to Business Intelligence (BI)

* Reporting
* OLAP Tools
* Visualisation
* Typically working with data in data warehouse

#### LEARN MORE
  
> "You can think of Business Intelligence as the process of going from raw data to legible information"
>
> [What is Business Intelligence](https://youtu.be/LFnewuBsYiY) (5 mins)


> "When IT professionals talk about Business Intelligence, they usually mean the analysis and presentation of data stored in an OLAP cube."
>
> [What is Business Intelligence and an OLAP Cube?](https://youtu.be/yoE6bgJv08E) (10 mins)

<img src="ConceptualMap-Emerging.png"></img>

## Emerging BI

* near real time
* behavioural data
* multi-disciplinary
* global connected world
* innovation and digital experimentation

#### LEARN MORE

> "The future of marketing, you're holding it in your hand already. Access to real time behavioural data will enable interaction between the brand and people.
>
> [World Business Trends 2017 | BI Business Review](https://youtu.be/Jas07US278M) (2 mins)

> "It is so critical that you can quickly answer questions and don't have to go back to IT and wait a few months"
>
>[What is a Modern BI Platform](https://youtu.be/1BaFnSOqJlQ) (3 mins)


#### LEARN MORE

> "People from multiple disciplines can actually make sense of data in a much more powerful way"
>
>[2018 Business Intelligence Trends](https://youtu.be/HqvZ3w1yYcE) (2 mins)

> "Maximising and optimising the connections between people and processes and technology"
>
> [Gartner Top 10 Strategic Technology Trends 2018](https://youtu.be/TPbKyD2bAR4) (6 mins)

<img src="ConceptualMap.png"></img>

### Main Theme: Addressing business concerns through storytelling with information

## Key focus areas:

1. **CONCERN:** The business concern or problem understood in the context of the business and relation to the stakeholders.

2. **DATA:** Potential sources of information that exist inside or outside of the business or which may be synthesised in order to address a business concern.

3. **ANALYTICS:** Techniques and processes and tools which can be utilised in analysing available data for the purposes of addressing a business concern.

4. **INTELLIGENCE:** Relationships, perspectives, narratives, and understandings that are supported by the data analytics in a way that is meaningful for stakeholders and holds efficacy in addressing a business concern.

### CONCERN

* what kind of problem - is it a business problem?
* who are the stakeholders?
* what is the context?
* business model disruption
* talent management
* global market trends
* foresight
* political risk


#### LEARN MORE

> "If you aren't harnessing the power of data, you're almost certain to end up falling behind."
>
> [The Top Issues CEOs Face These Days (2014)](https://www.wsj.com/articles/executive-leadership-what-are-the-top-issues-ceos-face-these-days-1395267060)

> "Don't ever try and present a technology solution to a business problem"
>
> [Technology Solutions Do Not Always Solve Business Problems](https://youtu.be/J7XAFa4wXgY)


#### LEARN MORE

> "Competitive innovation waits for no one"
>
> [Worst Company Disasters! | Top 6 Blunders](https://youtu.be/T0Z73Zbtlyg) (16 mins)

> "You promised me Mars colonies. Instead, I got Facebook."
>
> [Jason Pontin: Can technology solve our big problems?](https://youtu.be/ZB50BfYlsDc)

### DATA

* external vs internal
* external data for a bigger picture
* industry, consumer, product trends
* needs to be available for decisions
* quality an issue
* governance
* realtime


#### LEARN MORE

> "Only one-third of enterprises currently use information to identify new business opportunities and predict future trends and behavior"
>
> [14 Survey-Based Recommendations on How to Improve Data-Driven Decision-Making](https://bi-survey.com/data-driven-decision-making-business)

> "External data can give you real-time, minute-by-minute updates on industry, consumer, and product trends."
>
> [Why now is the perfect time to go all in on external data analytics](https://www.import.io/post/why-now-is-the-perfect-time-to-go-all-in-on-external-data-analytics/)

#### LEARN MORE

>"external data is one of the biggest blind spots in executive decision making today"
>
> [Outside Insight: Why External Data Is The Fuel Of Tomorrow's Business Success](https://www.forbes.com/sites/bernardmarr/2017/11/15/outside-insight-why-external-data-is-the-fuel-of-tomorrows-business-success/#443d8fa25e1d)

> "There are many free, external data sources posted around the Internet that can, if used well, completely transform our understanding of our market, audience, and the way we do business."
>
> [Free Data Sources to Upgrade Your Business Decision-Making](https://www.sisense.com/blog/free-data-sources-upgrade-business-decision-making/)

### ANALYTICS

* anchored to business value
* pragmatic approach
* test strategies
* invest in data for analytics insights


#### LEARN MORE

> "big data analytics is not trawl fishing. It’s spear fishing"
>
>[Big data analytics should be driven by business needs, not technology](https://www.mckinsey.com/business-functions/mckinsey-analytics/our-insights/big-data-analytics-should-be-driven-by-business-needs-not-technology)

> "So, one good rule of thumb is to always have a clear analytical objective."
> 
> [Here Are The Benefits of Data-Driven Decision Making](https://www.entrepreneur.com/article/280923)

### INTELLIGENCE

* proactivity
* mitigating risk
* customer experience
* design thinking for human problems

#### LEARN MORE

> "Today, businesses can collect data along every point of the customer journey"
>
>[5 Big Benefits of Data and Analytics for Positive Business Outcomes](http://blogs.teradata.com/data-points/5-big-benefits-data-analytics-positive-business-outcomes/)

> "In design, we build our way forward"
>
> [Want to Make Better Decisions? Know the Difference between Engineering and Design Thinking](https://youtu.be/q7LRxKHdao8) (7 mins)

